third_party/txt/src/txt/paragraph_txt.cc

```diff
@@ -429,6 +433,15 @@ bool ParagraphTxt::ComputeBidiRuns(std::vector<BidiRun>* result) {
       if (u_hasBinaryProperty(last_char, UCHAR_WHITE_SPACE)) {
         has_trailing_whitespace = true;
         bidi_run_count--;
+        // Check if the trailing whitespace occurs before the previous run or
+        // not. If so, this trailing whitespace was a leading whitespace.
+        int32_t second_last_bidi_run_start, second_last_bidi_run_length;
+        ubidi_getVisualRun(bidi.get(), bidi_run_count - 2,
+                           &second_last_bidi_run_start,
+                           &second_last_bidi_run_length);
+        if (bidi_run_start <= second_last_bidi_run_start) {
```

Comment:
There are two cases:
  1) last_bidi_run_start == second_last_bidi_run_end + 1
  2) last_bidi_run_start == second_last_bidi_run_start - 1

For #1, it's safe to merge the last run into the second-to-last run.

For #2, the `is_leading` flag will merge the last two runs to create a run where the last whitespace occurs at the start of the final run.  Is that desirable?  Or would we be better off leaving the bidi runs unmodified in this case?

Reply:
Let me try both modifying and not to see which one we want in terms of text input.

Reply:
Ok, it looks like both produce reasonable results during text input. In this case, I will prefer not modifying it when it is leading.